File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/data/network/api/MaintenanceApi.java
package com.lattis.ellipse.data.network.api;


import com.lattis.ellipse.data.network.model.body.bike.BikeDetailBody;
import com.lattis.ellipse.data.network.model.body.maintenance.DamageBikeBody;
import com.lattis.ellipse.data.network.model.response.BasicResponse;

import retrofit2.http.Body;
import retrofit2.http.POST;
import retrofit2.http.PUT;
import io.reactivex.Observable;

/**
 * Created by Velo Labs Android on 05-04-2017.
 */

public interface MaintenanceApi {
    @POST("maintenance/create-damage-report/")
    Observable<BasicResponse> damageBikes(@Body DamageBikeBody damageBikeBody);

    @PUT("maintenance/report-bike-theft/")
    Observable<BasicResponse> reportBikeTheft(@Body BikeDetailBody BikeDetailBody);




}

File: velolabs_repos/lattis_android_v2/data/src/main/java/com/lattis/data/repository/datasources/api/maintenance/MaintenanceApi.kt
package com.lattis.data.repository.datasources.api.maintenance

import com.lattis.data.entity.body.bike.BikeDetailBody
import com.lattis.data.entity.body.maintenance.DamageBikeBody
import com.lattis.data.entity.response.BasicResponse
import io.reactivex.rxjava3.core.Observable
import retrofit2.http.Body
import retrofit2.http.POST
import retrofit2.http.PUT

interface MaintenanceApi {
    @POST("api/maintenance/create-damage-report/")
    fun damageBikes(@Body damageBikeBody: DamageBikeBody): Observable<BasicResponse>

    @PUT("api/maintenance/report-bike-theft/")
    fun reportBikeTheft(@Body BikeDetailBody: BikeDetailBody): Observable<BasicResponse>
}